src/libraries/LiquidatorLib.sol

```diff
@@ -296,4 +355,172 @@ library LiquidatorLib {
       _minK
     );
   }
+
+  /**
+   * @notice Calculates the maximum amount of tokens in that results in the desired price impact.
+   * @param reserve1 The current reserve of token 1
+   * @param reserve0 The curernt reserve of token 0
+   * @param priceImpact1 The price impact to allow after swapping in the resulting amount
+   * @return amountIn1 The amount of token 1 to swap in to achieve the desired maximum price impact
+   */
+  function calculateRestrictedAmounts(
+    uint128 reserve1,
+    uint128 reserve0,
+    UFixed32x4 priceImpact1
+  ) internal pure returns (uint256 amountIn1, uint256 amountOut0) {
+    // If the price of token 1 is really small, do a different calculation to avoid rounding errors.
+    if (reserve1 > reserve0) {
+      return calculateRestrictedAmountsInverted(reserve1, reserve0, priceImpact1);
+    }
+
+    uint256 scalar = reserve0 - reserve1 >= type(uint88).max ? 1 : 1e8;
```

Comment:
If the delta between the reserves is > `type(uint88).max` then the "price" is going to be a very large number. If the "price" is scaled up at that point, `uint256(reserve0) * reserve1 * price1_1` won't fit inside a uint256 so we conditionally scale. 

Since `reserve0 > reserve1` by a large amount, we don't need the additional precision scaling because we will get a whole number for this value.